pkg/streamingpromql/query.go

```diff
@@ -332,8 +340,7 @@ func (q *Query) populateMatrixFromInstantVectorOperator(ctx context.Context, o o
 
 		if len(d.Floats) == 0 && len(d.Histograms) == 0 {
 			operator.PutFPointSlice(d.Floats)
-			// TODO: put histogram point slice back in pool
-
+			operator.PutHPointSlice(d.Histograms)
```

Comment:
We should be able to use `returnSeriesDataSlices` here as well.

Reply:
good catch, thankyou